### ISSUE TITLE
Question about exchange.createMarketOrder() . 

### ISSUE THREAD

**Issue by karlandoh** [ccxt/ccxt#2547]
I bolded the issue.

- OS: El Capitan
- Programming Language version: Python
- CCXT version: Latest
- Exchange: Binance, Poloniex (could be more).
- Method:

binance.createMarketBuyOrder('XVG/BTC',0.02)

Request: POST https://api.binance.com/api/v3/order {'X-MBX-APIKEY': HIDDEN, 'Content-Type': 'application/x-www-form-urlencoded', 'User-Agent': 'python-requests/2.18.4', 'Accept-Encoding': 'gzip, deflate'} timestamp=1523450667677&recvWindow=5000&symbol=XVGBTC&**quantity=0**&type=MARKET&side=BUY&signature=HIDDEN

Response: POST https://api.binance.com/api/v3/order 400 {'Content-Type': 'application/json', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'Date': 'Wed, 11 Apr 2018 12:44:27 GMT', 'Server': 'nginx', 'Strict-Transport-Security': 'max-age=31536000; includeSubdomains', 'X-Frame-Options': 'SAMEORIGIN', 'X-Xss-Protection': '1; mode=block', 'X-Content-Type-Options': 'nosniff', 'Content-Security-Policy': "default-src 'self'", 'X-Content-Security-Policy': "default-src 'self'", 'X-WebKit-CSP': "default-src 'self'", 'Cache-Control': 'no-cache, no-store, must-revalidate', 'Pragma': 'no-cache', 'Expires': '0', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 20f7b0e424b1512afc6d82d3d0bb8ab4.cloudfront.net (CloudFront)', 'X-Amz-Cf-Id': '18zXGdQ5kxh1uY6gLIYE0RqndmbPaKaKKVxCCjvfJv3FKkIsYVgGHA=='} {"code":-1013,"msg":"Invalid quantity."}

Now I feel like it would be better for the exchange to reference the balance I already have, instead. Does anyone have comments about this? This has also been the case on poloniex as well. Is there something that I can do instead?

Thanks.

**Comment by kroitor** [comment]
> Does anyone have comments about this? 

Yes, for XVG/BTC the minimal order amount is 1 XVG. You can't order 0.02 of XVG.

<img width="890" alt="screen shot 2018-04-11 at 15 56 02" src="https://user-images.githubusercontent.com/1294454/38617918-e27756f0-3da0-11e8-9347-11dddaf2cecb.png">

This is documented here: https://github.com/ccxt/ccxt/wiki/Manual#precision-and-limits

> This has also been the case on poloniex as well. 

Each market (symbol) of each exchange has its own specific trading limits, you should follow the rules of the exchange.

> Is there something that I can do instead?

Yep, read the Manual from the very beginning to the very end ;)

Let us know if this does not answer your question. Thx!

**Comment by karlandoh** [comment]
That's not what I'm asking..

I don't want to buy 0.02 XVG. I want to buy 0.02BTC WORTH of BTC. 

Thanks.

**Comment by kroitor** [comment]
@karlandoh in general, when you are trading XVG/BTC you are:

1. either selling XVG for BTC
2. or buying XVG for BTC

You can't buy BTC for BTC )

So, which one you want, 1 or 2 ? )

**Comment by karlandoh** [comment]
Edited

**Comment by karlandoh** [comment]
Sorry for typo

**Comment by kroitor** [comment]
Ok, when you want to buy 0.02BTC worth of some amount of XVG, it's exactly the same as selling XVG worth of 0.02BTC. However, with most exchanges you can't specify how much of BTC you want to spend, instead, you should specify how much of XVG you want to sell.

**Comment by kroitor** [comment]
Let me know if the above does not answer your question. Thank you!

**Comment by karlandoh** [comment]
I apologize. Maybe I am not clear.

I have 0.02BTC. I do not have any XVG. I want to use my 0.02BTC to purchase 'x' amount of XVG.

The issue is, according to the function, I will need to know **before hand** how much XVG I would like to purchase. However, I just want to buy 0.02BTC worth of XVG. That is all I currently know, and want to do.

When I asked "is there something I can do Instead" I meant:
 - Will I need to whip up a 'conversion factor' in order to determine how much XVG I would like to purchase?

**Comment by karlandoh** [comment]
Oh nice that is exactly what I was eluding to! Thanks again!

**Comment by kroitor** [comment]
@karlandoh be careful with the price that you choose, because the price is up to you... The answer to your question is: yes, you will have to know the price and amount beforehand, if you want to spend all bitcoins, but CCXT has tools for you to get the conversion rates from currently traded actual prices (orderbook for detailed analysis, ticker for brief inspection).

**Comment by kroitor** [comment]
@karlandoh we have added the support for total cost for market orders with Binance here: https://github.com/ccxt/ccxt/issues/6391#issuecomment-576247330
